### PR TITLE
8294399: (ch) Refactor some methods out of sun.nio.ch.UnixFileDispatcherImpl

### PR DESCRIPTION
**Pull request by bplb**
Interpose a new class `sun.nio.ch.UnixDispatcher` between `sun.nio.ch.NativeDispatcher` and its current subclasses `sun.nio.ch.DatagramDispatcher` and `sun.nio.ch.SocketDispatcher`; refactor some methods out of `sun.nio.ch.UnixFileDispatcherImpl` into this updated hierarchy.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294399](https://bugs.openjdk.org/browse/JDK-8294399): (ch) Refactor some methods out of sun.nio.ch.UnixFileDispatcherImpl

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10434/head:pull/10434` \
`$ git checkout pull/10434`

Update a local copy of the PR: \
`$ git checkout pull/10434` \
`$ git pull https://git.openjdk.org/jdk pull/10434/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10434`

View PR using the GUI difftool: \
`$ git pr show -t 10434`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10434.diff">https://git.openjdk.org/jdk/pull/10434.diff</a>

</details>

### CODE REVIEW

**Comment by bplb** [comment]
Ping.

File: src/java.base/unix/native/libnio/ch/UnixDispatcher.c [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+JNIEXPORT void JNICALL
+Java_sun_nio_ch_UnixDispatcher_init(JNIEnv *env, jclass clazz)
+{
+    int sp[2];
+    if (socketpair(PF_UNIX, SOCK_STREAM, 0, sp) < 0) {
+        JNU_ThrowIOExceptionWithLastError(env, "socketpair failed");
+        return;
+    }
+    preCloseFD = sp[0];
+    close(sp[1]);
```
Comment (AlanBateman) [inline src/java.base/unix/native/libnio/ch/UnixDispatcher.c:68]: This looks okay, just surprising to see the init methods at the end of the .c files when the previous convention had them before the other JNI methods.

**Review by AlanBateman — APPROVED** [review]

File: src/java.base/unix/native/libnio/ch/UnixDispatcher.c (hunk 1 above)
Comment (bplb) [reply src/java.base/unix/native/libnio/ch/UnixDispatcher.c]: Fixed in ab83200d83ce853e45ee6b6addb7f4b1d400d53f.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**